src/query/v2/plan/operator.cpp

```diff
@@ -221,6 +224,10 @@ class DistributedCreateNodeCursor : public Cursor {
       rqst.label_ids.push_back(msgs::Label{.id = primary_label});
       requests.push_back(std::move(rqst));
     }
+    const auto &rqst = requests.front();
+    msgs::Vertex v{.id = std::make_pair(rqst.label_ids.front(), rqst.primary_key)};
+    frame[nodes_info_.front()->symbol] =
+        TypedValue(query::v2::accessors::VertexAccessor(std::move(v), rqst.properties, context.shard_request_manager));
```

Comment:
Nope, there is no need. We already know ALL the information of the soon to be created vertex and therefore we put it  on the frame. If the request fails, an exception will be thrown and we are safe. Otherwise, by the time we get a response the Vertex will already exist on the frame

Reply:
The function is called `NodeCreationInfoToRequest`, there I think it should only transfrom the node creation info to a request and nothing more. This is a very unexpected side effect of this function which is not wanted in my opinion. Furthermore, the two things (transforming and putting value on the frame) can be clearly separated.